In [2]:
import pandas as pd

### 최종 모델 데이터 가져옴 

In [3]:
df = pd.read_csv('data/모델돌릴꺼2.csv')
df.head(2)

Unnamed: 0      라인   win  분당 킬  분당 어시  분당 데스   퍼블유무  퍼블어시유무  분당 얻은 골드  \
0           0     TOP  True  0.54   0.11   0.22  False   False    509.12   
1           1  JUNGLE  True  0.14   0.69   0.14  False    True    382.95   

   분당 피해량  분당 cc건시간  분당 CS  분당 죽은시간  분당 시야점수  
0  862.20      0.98    5.0     5.65     0.36  
1  570.17      0.80    1.0     2.97     0.76

### Unnamed: 0 제거

In [4]:
df.drop('Unnamed: 0',1,inplace=True)
df.head(2)

C:\Users\sujong\AppData\Local\Temp\ipykernel_24824\3130116340.py:1: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.
  df.drop('Unnamed: 0',1,inplace=True)


라인   win  분당 킬  분당 어시  분당 데스   퍼블유무  퍼블어시유무  분당 얻은 골드  분당 피해량  \
0     TOP  True  0.54   0.11   0.22  False   False    509.12  862.20   
1  JUNGLE  True  0.14   0.69   0.14  False    True    382.95  570.17   

   분당 cc건시간  분당 CS  분당 죽은시간  분당 시야점수  
0      0.98    5.0     5.65     0.36  
1      0.80    1.0     2.97     0.76

In [5]:
df['라인'].unique()

array(['TOP', 'JUNGLE', 'MIDDLE', 'BOTTOM', 'UTILITY'], dtype=object)

### 각 라인별로 데이터 나누기

In [6]:
top = df.loc[df['라인']=='TOP']
jungle = df.loc[df['라인']=='JUNGLE']
mid = df.loc[df['라인']=='MIDDLE']
bottom = df.loc[df['라인']=='BOTTOM']
utility = df.loc[df['라인']=='UTILITY']

### 라인 구분할때 사용했던 컬럼 제거

In [7]:
top.drop(['라인'],1, inplace=True)
jungle.drop(['라인'],1, inplace=True)
mid.drop(['라인'],1, inplace=True)
bottom.drop(['라인'],1, inplace=True)
utility.drop(['라인'],1, inplace=True)

C:\Users\sujong\AppData\Local\Temp\ipykernel_24824\2549572576.py:1: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.
  top.drop(['라인'],1, inplace=True)
C:\Users\sujong\AppData\Local\Temp\ipykernel_24824\2549572576.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  top.drop(['라인'],1, inplace=True)
C:\Users\sujong\AppData\Local\Temp\ipykernel_24824\2549572576.py:2: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.
  jungle.drop(['라인'],1, inplace=True)
C:\Users\sujong\AppData\Local\Temp\ipykernel_24824\2549572576.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentat

### 탑의 승리와 가장 연관있는 지표 알아보기

In [8]:
top.corr()

win      분당 킬     분당 어시     분당 데스      퍼블유무    퍼블어시유무  \
win       1.000000  0.338310  0.336567 -0.442457  0.067262  0.050431   
분당 킬      0.338310  1.000000  0.140860 -0.176069  0.252319 -0.003316   
분당 어시     0.336567  0.140860  1.000000 -0.082527 -0.002509  0.112738   
분당 데스    -0.442457 -0.176069 -0.082527  1.000000 -0.026480 -0.049485   
퍼블유무      0.067262  0.252319 -0.002509 -0.026480  1.000000 -0.086728   
퍼블어시유무    0.050431 -0.003316  0.112738 -0.049485 -0.086728  1.000000   
분당 얻은 골드  0.444460  0.827638  0.168239 -0.303028  0.214478  0.023820   
분당 피해량    0.201623  0.709484  0.294190  0.028736  0.156963  0.009289   
분당 cc건시간  0.046946  0.149437  0.277897  0.001949  0.026921  0.027493   
분당 CS     0.129293  0.201767 -0.198751 -0.285629  0.053776  0.005091   
분당 죽은시간  -0.358863 -0.060889  0.015251  0.870565 -0.004600 -0.041852   
분당 시야점수   0.139445  0.125854  0.174969 -0.107345  0.032727  0.023640   

          분당 얻은 골드    분당 피해량  분당 cc건시간     분당 CS   분당 죽은시간   분당 시야점수  
win       0.444460  0.201623  0.046946  0.129293 -0.358863  0.139445  
분당 킬      0.827638  0.709484  0.149437  0.201767 -0.060889  0.125854  
분당 어시     0.168239  0.294190  0.277897 -0.198751  0.015251  0.174969  
분당 데스    -0.303028  0.028736  0.001949 -0.285629  0.870565 -0.107345  
퍼블유무      0.214478  0.156963  0.026921  0.053776 -0.004600  0.032727  
퍼블어시유무    0.023820  0.009289  0.027493  0.005091 -0.041852  0.023640  
분당 얻은 골드  1.000000  0.670014  0.063888  0.556252 -0.158799  0.200020  
분당 피해량    0.670014  1.000000  0.234509  0.200173  0.216587  0.185702  
분당 cc건시간  0.063888  0.234509  1.000000 -0.111211  0.031006  0.069441  
분당 CS     0.556252  0.200173 -0.111211  1.000000 -0.243296  0.120512  
분당 죽은시간  -0.158799  0.216587  0.031006 -0.243296  1.000000 -0.014304  
분당 시야점수   0.200020  0.185702  0.069441  0.120512 -0.014304  1.000000

탑의 경우 관련있는 상위 5개 항목   
- 분당 얻은 골드 0.444
- 분당 데스 -0.442
- 분당 죽은시간 -0.359
- 분당 킬 0.338
- 분당 어시 0.337

  
탑의 경우 관련있는 하위 5개 항목  
- 분당 cc건 시간 0.0469
- 퍼블 어시유무 0.0504
- 퍼블 유무 0.0673
- 분당 cs 0.129
- 분당 시야점수 0.139 

### 정글의 승리와 가장 연관있는 지표 알아보기

In [9]:
jungle.corr()

win      분당 킬     분당 어시     분당 데스      퍼블유무    퍼블어시유무  \
win       1.000000  0.329481  0.372398 -0.524755  0.053865  0.063986   
분당 킬      0.329481  1.000000  0.091682 -0.173425  0.155815  0.014106   
분당 어시     0.372398  0.091682  1.000000 -0.135044  0.005384  0.154181   
분당 데스    -0.524755 -0.173425 -0.135044  1.000000 -0.048833 -0.035949   
퍼블유무      0.053865  0.155815  0.005384 -0.048833  1.000000 -0.106002   
퍼블어시유무    0.063986  0.014106  0.154181 -0.035949 -0.106002  1.000000   
분당 얻은 골드  0.472286  0.865487  0.237797 -0.335931  0.140766  0.043565   
분당 피해량    0.189947  0.734447  0.285775  0.046231  0.095531  0.028472   
분당 cc건시간  0.033904  0.059185  0.220537 -0.003722  0.010886  0.018877   
분당 CS    -0.008600  0.196449  0.007843  0.007316  0.022881  0.008425   
분당 죽은시간  -0.442531 -0.073637 -0.058266  0.879009 -0.029615 -0.029916   
분당 시야점수   0.168032  0.102556  0.175515 -0.134933  0.037256  0.050242   

          분당 얻은 골드    분당 피해량  분당 cc건시간     분당 CS   분당 죽은시간   분당 시야점수  
win       0.472286  0.189947  0.033904 -0.008600 -0.442531  0.168032  
분당 킬      0.865487  0.734447  0.059185  0.196449 -0.073637  0.102556  
분당 어시     0.237797  0.285775  0.220537  0.007843 -0.058266  0.175515  
분당 데스    -0.335931  0.046231 -0.003722  0.007316  0.879009 -0.134933  
퍼블유무      0.140766  0.095531  0.010886  0.022881 -0.029615  0.037256  
퍼블어시유무    0.043565  0.028472  0.018877  0.008425 -0.029916  0.050242  
분당 얻은 골드  1.000000  0.703299  0.069750  0.324348 -0.205053  0.165466  
분당 피해량    0.703299  1.000000  0.111277  0.321852  0.218552  0.137090  
분당 cc건시간  0.069750  0.111277  1.000000  0.021512  0.027226  0.062897  
분당 CS     0.324348  0.321852  0.021512  1.000000  0.084029  0.050221  
분당 죽은시간  -0.205053  0.218552  0.027226  0.084029  1.000000 -0.081619  
분당 시야점수   0.165466  0.137090  0.062897  0.050221 -0.081619  1.000000

정글의 경우 관련있는 상위 5개 항목   
- 분당 데스 -0.525
- 분당 얻은 골드 0.472
- 분당 죽은시간 -0.442
- 분당 어시 0.372
- 분당 킬 0.329

  
정글의 경우 관련있는 하위 5개 항목 
- 분당 cs 0.0086
- 분당 cc건 시간 0.0339
- 퍼블 유무 0.0539
- 퍼블 어시유무 0.064
- 분당 시야점수0.168   

### 미드의 승리와 가장 연관있는 지표 알아보기

In [10]:
mid.corr()

win      분당 킬     분당 어시     분당 데스      퍼블유무    퍼블어시유무  \
win       1.000000  0.349515  0.378753 -0.446784  0.059787  0.054920   
분당 킬      0.349515  1.000000  0.130996 -0.151535  0.207215  0.005132   
분당 어시     0.378753  0.130996  1.000000 -0.123101 -0.008703  0.138551   
분당 데스    -0.446784 -0.151535 -0.123101  1.000000 -0.011397 -0.041761   
퍼블유무      0.059787  0.207215 -0.008703 -0.011397  1.000000 -0.088597   
퍼블어시유무    0.054920  0.005132  0.138551 -0.041761 -0.088597  1.000000   
분당 얻은 골드  0.442749  0.840860  0.203734 -0.270513  0.173874  0.031573   
분당 피해량    0.235702  0.727776  0.342679 -0.023473  0.132640  0.018582   
분당 cc건시간  0.048879  0.033480  0.305020 -0.036920 -0.004080  0.047920   
분당 CS     0.117358  0.128830 -0.114618 -0.313794  0.009995  0.004884   
분당 죽은시간  -0.371286 -0.062823 -0.030421  0.890093  0.004069 -0.031786   
분당 시야점수   0.168712  0.118975  0.205506 -0.143430  0.024021  0.030943   

          분당 얻은 골드    분당 피해량  분당 cc건시간     분당 CS   분당 죽은시간   분당 시야점수  
win       0.442749  0.235702  0.048879  0.117358 -0.371286  0.168712  
분당 킬      0.840860  0.727776  0.033480  0.128830 -0.062823  0.118975  
분당 어시     0.203734  0.342679  0.305020 -0.114618 -0.030421  0.205506  
분당 데스    -0.270513 -0.023473 -0.036920 -0.313794  0.890093 -0.143430  
퍼블유무      0.173874  0.132640 -0.004080  0.009995  0.004069  0.024021  
퍼블어시유무    0.031573  0.018582  0.047920  0.004884 -0.031786  0.030943  
분당 얻은 골드  1.000000  0.695353  0.059479  0.486112 -0.157257  0.191861  
분당 피해량    0.695353  1.000000  0.218549  0.140593  0.145003  0.189781  
분당 cc건시간  0.059479  0.218549  1.000000  0.007713 -0.002977  0.101278  
분당 CS     0.486112  0.140593  0.007713  1.000000 -0.293659  0.128613  
분당 죽은시간  -0.157257  0.145003 -0.002977 -0.293659  1.000000 -0.057367  
분당 시야점수   0.191861  0.189781  0.101278  0.128613 -0.057367  1.000000

미드의 경우 관련있는 상위 5개 항목   
- 분당 데스 -0.447
- 분당 얻은 골드 0.443
- 분당 어시 0.379
- 분당 죽은시간-0.371
- 분당 킬 0.35


  
미드의 경우 관련있는 하위 5개 항목 
- 분당 cc건 시간 0.0489
- 퍼블 어시유무 0.0549
- 퍼블 유무 0.0598
- 분당 cs 0.117
- 분당 시야점수0.169   

### 원딜의 승리와 가장 연관있는 지표 알아보기

In [11]:
bottom.corr()

win      분당 킬     분당 어시     분당 데스      퍼블유무    퍼블어시유무  \
win       1.000000  0.380219  0.394614 -0.442942  0.072449  0.056484   
분당 킬      0.380219  1.000000  0.224999 -0.140111  0.233897  0.058157   
분당 어시     0.394614  0.224999  1.000000 -0.084794  0.020853  0.152103   
분당 데스    -0.442942 -0.140111 -0.084794  1.000000 -0.020860 -0.022766   
퍼블유무      0.072449  0.233897  0.020853 -0.020860  1.000000 -0.121720   
퍼블어시유무    0.056484  0.058157  0.152103 -0.022766 -0.121720  1.000000   
분당 얻은 골드  0.464847  0.883180  0.319448 -0.250246  0.213682  0.080334   
분당 피해량    0.244148  0.742271  0.377798 -0.016456  0.151513  0.061160   
분당 cc건시간  0.044521  0.069788  0.301399 -0.002438  0.014661  0.053470   
분당 CS     0.105824  0.184387 -0.065257 -0.327565  0.050875  0.020014   
분당 죽은시간  -0.343941 -0.032807  0.010648  0.860392  0.005637 -0.010612   
분당 시야점수   0.172455  0.133735  0.221399 -0.128017  0.041016  0.039391   

          분당 얻은 골드    분당 피해량  분당 cc건시간     분당 CS   분당 죽은시간   분당 시야점수  
win       0.464847  0.244148  0.044521  0.105824 -0.343941  0.172455  
분당 킬      0.883180  0.742271  0.069788  0.184387 -0.032807  0.133735  
분당 어시     0.319448  0.377798  0.301399 -0.065257  0.010648  0.221399  
분당 데스    -0.250246 -0.016456 -0.002438 -0.327565  0.860392 -0.128017  
퍼블유무      0.213682  0.151513  0.014661  0.050875  0.005637  0.041016  
퍼블어시유무    0.080334  0.061160  0.053470  0.020014 -0.010612  0.039391  
분당 얻은 골드  1.000000  0.744281  0.056784  0.450322 -0.104060  0.223431  
분당 피해량    0.744281  1.000000  0.168895  0.193511  0.166789  0.204720  
분당 cc건시간  0.056784  0.168895  1.000000 -0.075707  0.022402  0.147414  
분당 CS     0.450322  0.193511 -0.075707  1.000000 -0.290928  0.127177  
분당 죽은시간  -0.104060  0.166789  0.022402 -0.290928  1.000000 -0.017725  
분당 시야점수   0.223431  0.204720  0.147414  0.127177 -0.017725  1.000000

원딜의 경우 관련있는 상위 5개 항목  
- 분당 얻은 골드 0.465 
- 분당 데스 -0.443
- 분당 어시 0.395
- 분당 킬 0.380
- 분당 죽은시간-0.344

  
원딜의 경우 관련있는 하위 5개 항목 
- 분당 cc건 시간 0.0445
- 퍼블 어시유무 0.0565
- 퍼블 유무 0.0724
- 분당 cs 0.105
- 분당 시야점수 0.172   

### 서폿의 승리와 가장 연관있는 지표 알아보기

In [12]:
utility.corr()

win      분당 킬     분당 어시     분당 데스      퍼블유무    퍼블어시유무  \
win       1.000000  0.204594  0.490417 -0.443343  0.041053  0.088492   
분당 킬      0.204594  1.000000  0.040995 -0.005496  0.169989  0.036135   
분당 어시     0.490417  0.040995  1.000000 -0.212927  0.024557  0.163661   
분당 데스    -0.443343 -0.005496 -0.212927  1.000000 -0.017941 -0.041645   
퍼블유무      0.041053  0.169989  0.024557 -0.017941  1.000000 -0.111121   
퍼블어시유무    0.088492  0.036135  0.163661 -0.041645 -0.111121  1.000000   
분당 얻은 골드  0.438192  0.838293  0.345290 -0.127083  0.146254  0.107071   
분당 피해량    0.108316  0.648871  0.092282  0.115687  0.074205  0.013661   
분당 cc건시간  0.020564  0.117704  0.213502  0.095153  0.030248  0.064712   
분당 CS    -0.028547  0.316490 -0.155005  0.107126  0.042490  0.006105   
분당 죽은시간  -0.364059  0.066115 -0.111733  0.865719 -0.004318 -0.021517   
분당 시야점수   0.168077  0.058181  0.279521 -0.084900  0.020697  0.048912   

          분당 얻은 골드    분당 피해량  분당 cc건시간     분당 CS   분당 죽은시간   분당 시야점수  
win       0.438192  0.108316  0.020564 -0.028547 -0.364059  0.168077  
분당 킬      0.838293  0.648871  0.117704  0.316490  0.066115  0.058181  
분당 어시     0.345290  0.092282  0.213502 -0.155005 -0.111733  0.279521  
분당 데스    -0.127083  0.115687  0.095153  0.107126  0.865719 -0.084900  
퍼블유무      0.146254  0.074205  0.030248  0.042490 -0.004318  0.020697  
퍼블어시유무    0.107071  0.013661  0.064712  0.006105 -0.021517  0.048912  
분당 얻은 골드  1.000000  0.640848  0.148850  0.356177 -0.031904  0.158585  
분당 피해량    0.640848  1.000000  0.268212  0.416742  0.251254  0.154495  
분당 cc건시간  0.148850  0.268212  1.000000  0.129667  0.128848  0.135418  
분당 CS     0.356177  0.416742  0.129667  1.000000  0.155889  0.028014  
분당 죽은시간  -0.031904  0.251254  0.128848  0.155889  1.000000  0.034280  
분당 시야점수   0.158585  0.154495  0.135418  0.028014  0.034280  1.000000

서폿의 경우 관련있는 상위 5개 항목   
- 분당 어시 0.490
- 분당 데스 -0.443
- 분당 얻은 골드 0.438
- 분당 죽은시간-0.364
- 분당 킬 0.204


  
서폿의 경우 관련있는 하위 5개 항목 
- 분당 cc건 시간 0.0206
- 분당 cs -0.0285
- 퍼블 유무 0.0411
- 퍼블 어시유무 0.0885
- 분당 시야점수 0.168   